device_doctor/lib/src/ios_device.dart

```diff
@@ -76,6 +78,39 @@ class IosDeviceDiscovery implements DeviceDiscovery {
     }
     return healthCheckResult;
   }
+
+  Future<HealthCheckResult> certCheck({ProcessManager processManager}) async {
+    HealthCheckResult healthCheckResult;
+    try {
+      final String certCheckResult =
+          await eval('security', <String>['find-identity', '-p', 'codesigning', '-v'], processManager: processManager);
+      if (certCheckResult.contains('Apple Development: Flutter Devicelab') &&
+          certCheckResult.contains('1 valid identities found')) {
+        healthCheckResult = HealthCheckResult.success(kCertCheckKey);
+      } else {
+        healthCheckResult = HealthCheckResult.failure(kCertCheckKey, certCheckResult);
+      }
+    } on BuildFailedError catch (error) {
+      healthCheckResult = HealthCheckResult.failure(kCertCheckKey, error.toString());
+    }
+    return healthCheckResult;
+  }
+
+  Future<HealthCheckResult> devicePairCheck({ProcessManager processManager}) async {
+    HealthCheckResult healthCheckResult;
+    try {
+      final String devicePairCheckResult =
+          await eval('idevicepair', <String>['validate'], processManager: processManager);
+      if (devicePairCheckResult.contains('SUCCESS')) {
+        healthCheckResult = HealthCheckResult.success(kDevicePairCheckKey);
+      } else {
+        healthCheckResult = HealthCheckResult.failure(kDevicePairCheckKey, devicePairCheckResult);
+      }
+    } on BuildFailedError catch (error) {
```

Comment:
Is BuildFailedError the only exception raised by eval?

Reply:
Yes. We have wrapped any error there as BuildFailedError.